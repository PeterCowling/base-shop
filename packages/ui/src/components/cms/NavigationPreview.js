import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React from "react";
export default function NavigationPreview({ items, style }) {
    return (_jsx("nav", { style: style, className: "bg-background text-foreground p-4 rounded border", "data-token": "--color-bg", children: _jsx("ul", { className: "flex gap-4", children: items.map((item) => (_jsx(NavItemView, { item: item }, item.id))) }) }));
}
function NavItemView({ item }) {
    return (_jsxs("li", { className: "relative group", children: [_jsx("a", { href: item.url || "#", className: "font-medium", "data-token": "--color-fg", children: item.label || "Item" }), item.children && item.children.length > 0 && (_jsx("ul", { className: "absolute left-0 top-full hidden min-w-[8rem] flex-col rounded-md border bg-background p-2 shadow-md group-hover:flex", "data-token": "--color-bg", children: item.children.map((child) => (_jsx("li", { children: _jsx("a", { href: child.url || "#", className: "block px-2 py-1 hover:underline", "data-token": "--color-fg", children: child.label || "Item" }) }, child.id))) }))] }));
}
