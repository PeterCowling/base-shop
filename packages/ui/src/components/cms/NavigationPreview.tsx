import React, { forwardRef } from "react";
import type { NavItem } from "./NavigationEditor";
// Use semantic lists to match tests and expected markup
import { useTranslations } from "@acme/i18n";
import { Inline, Stack } from "../atoms/primitives";

interface Props {
  items: NavItem[];
  style?: React.CSSProperties;
}

const StyledNav = forwardRef<HTMLElement, React.HTMLAttributes<HTMLElement>>(
  ({ children, ...rest }, ref) => (
    <nav ref={ref} {...rest}>
      {children}
    </nav>
  ),
);
StyledNav.displayName = "StyledNav";

export default function NavigationPreview({ items, style }: Props) {
  return (
    <StyledNav
      style={style}
      className="bg-surface-2 text-foreground p-4 rounded border border-border-1"
      data-token="--color-bg"
    >
      <Inline asChild gap={4} wrap={false} className="m-0 list-none p-0">
        <ul>
          {items.map((item) => (
            <NavItemView key={item.id} item={item} />
          ))}
        </ul>
      </Inline>
    </StyledNav>
  );
}

function NavItemView({ item }: { item: NavItem }) {
  const t = useTranslations();
  return (
    <li className="relative group">
      <a
        href={item.url || "#"}
        className="inline-flex min-h-10 min-w-10 items-center justify-center rounded-md px-3 font-medium hover:underline focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary"
        data-token="--color-fg"
      >
        {item.label || (t("nav.itemFallback") as string)}
      </a>
      {item.children && item.children.length > 0 && (
        <Stack
          asChild
          gap={1}
          className="absolute top-full start-0 min-w-32 rounded-md border border-border-1 bg-surface-2 p-2 shadow-elevation-2 opacity-0 transition-opacity duration-150 ease-out pointer-events-none group-hover:pointer-events-auto group-hover:opacity-100"
          data-token="--color-bg"
        >
          <ul className="m-0 list-none p-0">
            {item.children.map((child) => (
              <li key={child.id}>
                <a
                  href={child.url || "#"}
                  className="inline-flex min-h-10 min-w-10 items-center rounded px-2 hover:underline focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary"
                  data-token="--color-fg"
                >
                  {child.label || (t("nav.itemFallback") as string)}
                </a>
              </li>
            ))}
          </ul>
        </Stack>
      )}
    </li>
  );
}
