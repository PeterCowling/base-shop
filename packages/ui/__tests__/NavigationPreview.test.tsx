import { render, screen, within } from "@testing-library/react";
import NavigationPreview from "../src/components/cms/NavigationPreview";

describe("NavigationPreview", () => {
  it("renders nested menu links", () => {
    const items = [
      { id: "home", label: "Home", url: "/home" },
      {
        id: "products",
        label: "Products",
        url: "/products",
        children: [
          { id: "hats", label: "Hats", url: "/products/hats" },
          { id: "shirts", label: "Shirts", url: "/products/shirts" },
        ],
      },
      {
        id: "about",
        label: "About",
        url: "/about",
        children: [{ id: "team", label: "Team", url: "/about/team" }],
      },
      {
        id: "misc",
        label: "",
        url: "",
        children: [{ id: "nameless", label: "", url: "" }],
      },
    ];

    const { container } = render(<NavigationPreview items={items} />);

    const productsLink = screen.getByRole("link", { name: "Products" });
    expect(productsLink).toHaveAttribute("href", "/products");
    const productsItem = productsLink.closest("li")!;
    expect(
      within(productsItem).getByRole("link", { name: "Hats" }),
    ).toHaveAttribute("href", "/products/hats");
    expect(
      within(productsItem).getByRole("link", { name: "Shirts" }),
    ).toHaveAttribute("href", "/products/shirts");

    const aboutLink = screen.getByRole("link", { name: "About" });
    const aboutItem = aboutLink.closest("li")!;
    expect(
      within(aboutItem).getByRole("link", { name: "Team" }),
    ).toHaveAttribute("href", "/about/team");

    const placeholderLink = screen.getAllByRole("link", { name: "Item" })[0];
    expect(placeholderLink).toHaveAttribute("href", "#");
  
    const nav = container.querySelector("nav");
    expect(nav).toHaveClass(
      "bg-surface-2",
      "text-foreground",
      "p-4",
      "rounded",
      "border",
      "border-border-1",
    );
  });
});

