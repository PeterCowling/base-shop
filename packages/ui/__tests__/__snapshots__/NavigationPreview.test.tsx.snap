// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationPreview renders nested menu links 1`] = `
<nav
  class="bg-background text-foreground p-4 rounded border"
  data-token="--color-bg"
>
  <ul
    class="flex gap-4"
  >
    <li
      class="relative group"
    >
      <a
        class="font-medium"
        data-token="--color-fg"
        href="/home"
      >
        Home
      </a>
    </li>
    <li
      class="relative group"
    >
      <a
        class="font-medium"
        data-token="--color-fg"
        href="/products"
      >
        Products
      </a>
      <ul
        class="absolute left-0 top-full hidden min-w-[8rem] flex-col rounded-md border bg-background p-2 shadow-md group-hover:flex"
        data-token="--color-bg"
      >
        <li>
          <a
            class="block px-2 py-1 hover:underline"
            data-token="--color-fg"
            href="/products/hats"
          >
            Hats
          </a>
        </li>
        <li>
          <a
            class="block px-2 py-1 hover:underline"
            data-token="--color-fg"
            href="/products/shirts"
          >
            Shirts
          </a>
        </li>
      </ul>
    </li>
    <li
      class="relative group"
    >
      <a
        class="font-medium"
        data-token="--color-fg"
        href="/about"
      >
        About
      </a>
      <ul
        class="absolute left-0 top-full hidden min-w-[8rem] flex-col rounded-md border bg-background p-2 shadow-md group-hover:flex"
        data-token="--color-bg"
      >
        <li>
          <a
            class="block px-2 py-1 hover:underline"
            data-token="--color-fg"
            href="/about/team"
          >
            Team
          </a>
        </li>
      </ul>
    </li>
    <li
      class="relative group"
    >
      <a
        class="font-medium"
        data-token="--color-fg"
        href="#"
      >
        Item
      </a>
      <ul
        class="absolute left-0 top-full hidden min-w-[8rem] flex-col rounded-md border bg-background p-2 shadow-md group-hover:flex"
        data-token="--color-bg"
      >
        <li>
          <a
            class="block px-2 py-1 hover:underline"
            data-token="--color-fg"
            href="#"
          >
            Item
          </a>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
